### PR TITLE
Ben

### PR DESCRIPTION
**Pull request by Conundrum4**
Code review and test before merge with master branch

### CODE REVIEW

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+                if (grid[x, y + 2] == false)
+                {
+                    grid[x, y + 2] = true;
+                    grid[x, y + 1] = true;
+
+                    MazeDigger(x, y + 2);
+                }
+            }
+        }
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/Maze.cs]: 1. Please use a switch statement as you are checkinng the same variable with every if statement
2. Please add functions to replace these blocks of code. E.g. "CreateLeft(x,y)

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    public Vector3 GetGoalPosition()
+    {
+        int radius = 2;
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/Maze.cs]: Use `const` as this variable does not and should not change here after definition

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            for (int y = endY - radius; y <= endY + radius; y++)
+            {
+                if (GetCell(x, y))
+                {
+                    return new Vector3(x, y);
+                }
+            }
+        }
+
+        return Vector3.one * 1000;
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/Maze.cs]: It isnt clear what this denotes. please assign to an appropriately named variable such as `hiddenGoal`

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    public bool GetCell(int x, int y)
+    {
+        if (x >= width || x < 0 || y >= height || y <= 0)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/Maze.cs]: At brackets around each boolean statement

File: mazegame/Assets/Scripts/Effects/LightFlicker.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        while (true)
+        {
+            flashlight.intensity = lightIntensity;
+
+            float lightingTime = lightTime + ((float)rg.NextDouble() - 0.5f);
+            yield return new WaitForSeconds(lightingTime);
+
+            int flickerCount = rg.Next(4, 9);
+
+            for (int i = 0; i < flickerCount; i++)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Effects/LightFlicker.cs]: Use `var` here for the type

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+    public delegate void MazeReadyAction();
+    public static event MazeReadyAction OnMazeReady;
+
+
+    void Start()
+    {
+        mazeRG = new System.Random(mazeSeed.GetHashCode());
+
+        if (mazeWidth % 2 == 0)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: brackets

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+            mazeWidth++;
+
+        if (mazeHeight % 2 == 0)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: brackets

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+        if (OnMazeReady != null)
+        {
+            OnMazeReady();
+        }
+    }
+
+    void DrawMaze()
+    {
+        for (int x = 0; x < mazeWidth; x++)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Use `var` for type

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+        {
+            for (int y = 0; y < mazeHeight; y++)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Use var for type

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+            {
+                Vector3 position = new Vector3(x, y);
+
+                if (maze.Grid[x, y] == true)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs:63]: a comment here would be good. Not 100% sure what is being checked. Is it the edge of the maze? A corner?

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+        mazeSprite.transform.SetParent(parent);
+        mazeSprite.transform.Rotate(0, 0, rotation);
+    }
+
+    void DrawWalls(int x, int y)
+    {
+        bool top = GetMazeGridCell(x, y + 1);
+        bool bottom = GetMazeGridCell(x, y - 1);
+        bool right = GetMazeGridCell(x + 1, y);
+        bool left = GetMazeGridCell(x - 1, y);
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs:91]: all of these should be `const`

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+
+        if (!right && !bottom && x < mazeWidth - 1 && y > 0)
+        {
+            CreateMazeSprite(position, wallCornerSprite, transform, 2, 180);
+        }
+
+        if (!right && !top && x < mazeWidth - 1 && y < mazeHeight - 1)
+        {
+            CreateMazeSprite(position, wallCornerSprite, transform, 2, 270);
+        }
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Use brackets here where you can with the conditions. e.g. `if(!left && !top && (x > 0) && (y < mazeHeight-1))`

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+    void Awake()
+    {
+        instance = this;
+    }
+
+    public List<Vector3> GetRandomFloorPositions(int count)
+    {
+        List<Vector3> positions = new List<Vector3>();
+
+        for (int i = 0; i < count; i++)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Use `var` here for type

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+    public static MazeGenerator instance;
+
+    void Awake()
+    {
+        instance = this;
+    }
+
+    public List<Vector3> GetRandomFloorPositions(int count)
+    {
+        List<Vector3> positions = new List<Vector3>();
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: `floorPositions` would be a more appropriate name

File: mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+    int currentX = 1;
+    int currentY = 1;
+
+    float currentAngle;
+    float lastAngle;
+
+    void Update()
+    {
+        bool targetReached = transform.position.x == targetX && transform.position.y == targetY;
```
Comment (jrome5) [inline mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs]: Brackets

File: mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+
+        currentX = Mathf.FloorToInt(transform.position.x);
+        currentY = Mathf.FloorToInt(transform.position.y);
+
+        direction.x = Input.GetAxisRaw("Horizontal");
+        direction.y = Input.GetAxisRaw("Vertical");
+
+        float angle = 0;
```
Comment (jrome5) [inline mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs]: Please use a more appropriate name for this. What does this angle belong to

File: mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+            if (MazeGenerator.instance.GetMazeGridCell(currentX, currentY - 1) && targetReached)
+            {
+                targetX = currentX;
+                targetY = currentY - 1;
+            }
+        }
+        else
+        {
+            angle = lastAngle;
+        }
```
Comment (jrome5) [inline mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs]: Please use a switch statement. Only one variable is being checked here

**Review by jrome5 — CHANGES_REQUESTED** [review]

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 18]
```diff
@@ -0,0 +1,151 @@
...
+                    MazeDigger(x - 2, y);
+                }
+            }
+
+            if (directions[i] == 3)
+            {
+                if (x + 2 >= width - 1)
+                    continue;
+
+                if (grid[x + 2, y] == false)
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/Maze.cs]: combine these if statements such as `if(((x+2) < (width -1)) && (grid[x + 2, y] == false))`

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 19]
```diff
@@ -0,0 +1,151 @@
...
+                }
+            }
+
+            if (directions[i] == 4)
+            {
+                if (y + 2 >= height - 1)
+                    continue;
+
+                if (grid[x, y + 2] == false)
+                {
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/Maze.cs]: combine if statements

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 20]
```diff
@@ -0,0 +1,151 @@
...
+                }
+            }
+
+            if (directions[i] == 2)
+            {
+                if (x - 2 <= 0)
+                    continue;
+
+                if (grid[x - 2, y] == false)
+                {
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/Maze.cs]: combine if statements

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 21]
```diff
@@ -0,0 +1,151 @@
...
+        Tools.Shuffle(directions, rg);
+
+        for(int i = 0; i < directions.Length; i++) {
+            if (directions[i] == 1)
+            {
+                if (y - 2 <= 0)
+                    continue;
+
+                if (grid[x, y - 2] == false)
+                {
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/Maze.cs]: combine if statements

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 1 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: - Merged if statements
- Set-up switch statements

**Review by jrome5 — CHANGES_REQUESTED** [review]

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 2 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Added

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 22]
```diff
@@ -0,0 +1,151 @@
...
+                }
+            }
+        }
+    }
+
+    public Vector3 GetGoalPosition()
+    {
+        int radius = 2;
+
+        int endX = width - startX;
```
[the remark this answers is not in the document]
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Added

File: mazegame/Assets/Scripts/Maze/Maze.cs [hunk 23]
```diff
@@ -0,0 +1,151 @@
...
+        int endY = height - startY;
```
[the remark this answers is not in the document]
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Added

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 3 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Magic number, I don't know why it's being * 1000

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 4 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Done

File: mazegame/Assets/Scripts/Effects/LightFlicker.cs (hunk 5 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Effects/LightFlicker.cs]: Changed

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 6 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Added

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 7 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Added

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 8 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Changed

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 9 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Changed

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 10 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs:63]: // Check if the cell is a path, if it isn't, add it.

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 11 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs:91]: Doesn't work

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 12 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Added

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 13 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: Changed

File: mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs (hunk 15 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs]: Added

File: mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs (hunk 16 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs]: I haven't changed this.

File: mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs (hunk 17 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs]: Multiple variables, can't do switch statement

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 18 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Done

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 19 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Done

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 20 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Done

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 21 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: Done

File: mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs (hunk 17 above)
Comment (jrome5) [reply mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs]: Combine if statements

File: mazegame/Assets/Scripts/Maze/MazeGenerator.cs (hunk 14 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/MazeGenerator.cs]: changed

File: mazegame/Assets/Scripts/Maze/Maze.cs (hunk 3 above)
Comment (Conundrum4) [reply mazegame/Assets/Scripts/Maze/Maze.cs]: fixed, created new variable name

**Pull request merged**